lib/controller.js

```diff
@@ -452,6 +455,165 @@ controller.connectToNetwork = function(opts) {
+            // If this is the top of the pile, it's most recent
+            if (i === 0) {
+              // Print when it was published
+              logs.basic('\t Version:', build.version, '\tPublished:', released.toLocaleString());
+            } else {
+              // For the rest just print the version
+              logs.basic('\t Version:', build.version);
+            }
+          }
+          // Finish up
+          return resolve();
+        }, reject);
+      // This is a standard update
+    } else {
+      // Fetch the latest version
+      var requestedVersion = 'latest';
+      // Unless a specific build was request
+      if (opts.build) {
+        // Then we'll fetch that build
+        requestedVersion = opts.build;
+      }
+
+      // Grab our Tessel
+      var tesselPromise = Tessel.get(opts);
+      var shaPromise;
+
+      // If we aren't forcing, we'll want to get the current SHA on Tessel 
+      if (!opts.force) {
+        // Once we have the Tessel
+        shaPromise = tesselPromise.then(function(tessel) {
+          // Figure out what commit SHA is running on it
+          return tessel.fetchCurrentBuildInfo()
+            // Once we have the current SHA
+            .then(function(currentSHA) {
+              // Figure out the human readable version for that code release
+              return updates.fetchBuildVersionforSHA(currentSHA);
+
+            });
+        });
+      }
+      // Otherwise, just resolve
+      else {
+        shaPromise = Promise.resolve();
+      }
+
+      // Once we have our Tessel and build version
+      return Promise.all([tesselPromise, shaPromise]).spread(function(tessel, currentVersion) {
+        // If it's not connected via USB, we can't update it
+        if (!tessel.usbConnection) {
+          return reject('Must have Tessel connected over USB to complete update. Aborting update.');
+        }
+
+        // If this Tessel isn't connected to the LAN
+        if (!tessel.lanConnection || tessel.lanConnection.authorized) {
+          // Warn the user that it's going to take a while
+          logs.warn('Warning: Could not detect Tessel over LAN. Updates will be much slower over USB.');
+        }
+
+        // If the update is forced or this build was requested
+        if (opts.build) {
+          // Fetch the requested build
+          return controller.updateTesselWithVersion(tessel, currentVersion, requestedVersion)
+            .then(resolve, reject);
+        } else {
+          // If they have requested the latest firmware
+          var verifiedVersion;
+          if (requestedVersion === 'latest') {
+            // Figure out what the version is
+            verifiedVersion = updates.fetchBuildList()
+              .then(function(builds) {
+                // Make sure we actually have builds
+                if (Object.keys(builds).length === 0) {
+                  return reject('Could not detect any builds on server.');
```

Comment:
When might this happen? Does this mean they're not reaching the server? Seems like a good place to have a more helpful error message.